trajopt_sco/include/trajopt_sco/solver_interface.hpp

```diff
@@ -148,5 +149,53 @@ std::ostream& operator<<(std::ostream&, const Cnt&);
 std::ostream& operator<<(std::ostream&, const AffExpr&);
 std::ostream& operator<<(std::ostream&, const QuadExpr&);
 
-ModelPtr createModel();
+class ConvexSolver
+{
+public:
+  enum Value {
```

Comment:
A name like `ModelType` seems to describe this better.

Reply:
Thanks for the review! Yes, I am actually also a bit confused of the meaning of model vs solver that is used in the codebase. With the addition of solver_utils, I also have the feeling that all these different "Models" could be encapsulated in a single `Model`, which then make use of different Convex Solvers to solve them. For instance, the Model could provide Eigen types for the objective function and constraints, and the solver would use them as best as possible. Well, these suggestions of mine are probably not for this PR. I can see the idea beind the fact that `ModelType` would be more fitting here though. What do you think @Levi-Armstrong?

Reply:
If model is used throughout then probably best to keep consistent.

Reply:
on it